go-controller/pkg/cni/cnishim.go

```diff
@@ -194,7 +199,106 @@ func (p *Plugin) CmdDel(args *skel.CmdArgs) error {
 }
 
 // CmdCheck is the callback for 'checking' container's networking is as expected.
-// Currently not implemented, so returns `nil`.
 func (p *Plugin) CmdCheck(args *skel.CmdArgs) error {
+	var err error
+
+	startTime := time.Now()
+	defer func() {
+		p.postMetrics(startTime, CNICheck, err)
+	}()
+
+	// read the config stdin args
+	conf, err := config.ReadCNIConfig(args.StdinData)
+	if err != nil {
+		return fmt.Errorf("invalid stdin args")
+	}
+	setupLogging(conf)
+	netns, err := ns.GetNS(args.Netns)
+	if err != nil {
+		return fmt.Errorf("failed to open netns %q: %v", args.Netns, err)
+	}
+	defer netns.Close()
+
+	if conf.IPAM.Type != "" {
+		// run the IPAM plugin and get back the config to apply
+		err = ipam.ExecCheck(conf.IPAM.Type, args.StdinData)
+		if err != nil {
+			return err
+		}
+	}
+
+	if err := version.ParsePrevResult(&conf.NetConf); err != nil {
+		return err
+	}
+	result, err := current.NewResultFromResult(conf.PrevResult)
+	if err != nil {
+		return err
+	}
+
+	var contMap current.Interface
```

Comment:
`contMap` seems like a poor name, it's not a map. I guess `intf` is better since this is what you called it in the function signature on L283?

Reply:
actually, I copied this name (and part of the code) from https://github.com/containernetworking/plugins/tree/master/plugins
@squeed, what do you suggest ?

Reply:
what the heck was I thinking with `contMap`. No clue!? It's a poor choice of name.

Reply:
Renamed it.